### ISSUE TITLE
Docker: OPENGROK_WEBAPP_CONTEXT doesn't have full effect

### ISSUE THREAD

**Issue by timschumi** [oracle/opengrok#3153]
**Describe the bug**

When starting a Docker container, OPENGROK_WEBAPP_CONTEXT does not have full effect and the webpage is not available at the chosen path.

**To Reproduce**

1. Set up Docker using [this](https://gist.githubusercontent.com/timschumi/b5bb13111efc7aa2195defcc482a5d61/raw/fc4f0acc6d8296c761b09df98e7a09931e4fff03/docker-compose.yml) docker-compose file.
2. Create `src`, `data` and `etc` folders in the same directory.
3. Start the container.
4. Access http://localhost:8080/opengrok
5. Receive a "File not found" page due to OpenGrok still using / as its root.

**Expected behavior**
OpenGrok fully honors the OPENGROK_WEBAPP_CONTEXT variable and shows the main page when accessing /opengrok (or whatever was set).

**Screenshots**
[Main page](https://user-images.githubusercontent.com/16820960/82769096-3cdf2880-9e33-11ea-8daa-ab73806e0e11.png) and [/opengrok page](https://user-images.githubusercontent.com/16820960/82769100-48caea80-9e33-11ea-936e-127e598b95a8.png) when setting OPENGROK_WEBAPP_CONTEXT="/opengrok". The port has just been used during my testing, the linked docker-compose.yml launches the service on the default port 8080.

**Additional context**

I've tested that the OPENGROK_WEBAPP_CONTEXT variable is successfully propagated to the application. The issue rather is that [in the Dockerfile](https://github.com/oracle/opengrok/blob/52fc2931f25089055b5cf1dea11f846521d748ef/Dockerfile#L75), the OpenGrok .war file is explicitly installed as ROOT.war. Hardcoding that to something different moves the application to the corresponding directory (but this obviously isn't very portable and the Dockerfile isn't intended for that). One solution would be to move/link that file at runtime, but I'm unsure whether that is desired or how that should be done.

**Comment by vladak** [comment]
The `opengrok-deploy` command `RUN` should honor the `OPENGROK_WEBAPP_CONTEXT` variable. There is also a bit of a trickery done there with the redirect from `/source` to `/` , not sure this is necessary:

https://github.com/oracle/opengrok/blob/1c8c7ecb11894fe72ab40888ce85e2dcfb8752a4/Dockerfile#L72-L77

Also, [the Docker README](https://github.com/oracle/opengrok/tree/52fc2931f25089055b5cf1dea11f846521d748ef/docker#environment-variables) should document the variable.

PR to fix this is welcome.

**Comment by timschumi** [comment]
> The `opengrok-deploy` command `RUN` should honor the `OPENGROK_WEBAPP_CONTEXT` variable.

Problem is that the variable is only set by the user _after_ the build has finished (Image has been built, uploaded to and downloaded from the Docker Hub already). This means that while the variable is there during the build process, it only has its default value that is provided behind the `ENV` command.

Overriding the variable on the users end works, but at that point `opengrok-deploy` has been run already (and the result is baked firmly into the container image). To fix that, `opengrok-deploy` (or rather, the whole `RUN` chunk) would have to be run once during container start instead of during build-time.

> There is also a bit of a trickery done there with the redirect from `/source` to `/` , not sure this is necessary:

As far as I know, that was only a compatibility change when the default path was moved. I don't think that it will affect us much concerning this problem, except that it will have to be updated to point to the new directory (or rather, whatever the user chooses) as well.

> Also, [the Docker README](https://github.com/oracle/opengrok/tree/52fc2931f25089055b5cf1dea11f846521d748ef/docker#environment-variables) should document the variable.

In hindsight, it doesn't appear like this variable (while changeable by the user) is intended to be changed. Maybe it's indeed just there to make `opengrok-deploy` happy. I can also spot some other places where the path is simply hardcoded instead of relying on that variable.

> PR to fix this is welcome.

I fully intend to fix this myself, but I first wanted to try and get more information/suggestions about this, which is why I opened this issue first.
